### PR TITLE
Add & Remove Functions

### PR DESCRIPTION
**Pull request by rebaz36**
[To Do list: add & remove](https://github.com/microverseinc/curriculum-javascript/blob/main/todo-list/m3_add_remove.md)

- Update index.html
- Update styles
- Add Js. for todo functionality

### CODE REVIEW

File: src/todolist.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    }
+  }
+
+  // Methods
+  AddToDo(activity, doneBox) {
+    const id = this.#toDo.length;
+    this.#toDo.push({
+      activity,
+      doneBox,
+      id,
```
Comment (Kamba56) [inline src/todolist.js]: - [ ] Please asdemanded by the project requirements, the properties of each task should be `description`, `completed` and `index`. Kindly fix that please.

File: src/todolist.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+  reorder() {
+    for (let index = 0; index < this.#toDo.length; index += 1) {
+      this.#toDo[index].id = index;
+    }
+  }
+
+  // Methods
+  AddToDo(activity, doneBox) {
+    const id = this.#toDo.length;
```
Comment (Kamba56) [inline src/todolist.js]: - [ ] I noticed that your index is the length of the array before the new task is pushed in so your index starts from zero. Please as demanded by project requirements, kindly make the property `index` set to the value of the new array length (i.e. if you're adding a 5th task to the list, the index of that task should equal to 5). You an achieve this by changing line `24` of your code to `const index = this.#toDo.length + 1;`

File: src/index.js [hunk 3]
```diff
@@ -1,20 +1,97 @@
...
+    element.append(doneBox, inputTask);
+    doneBox.setAttribute('id', toDo.id);
+    inputTask.disabled = false;
+    doneBox.checked = toDo.doneBox;
+    if (doneBox.checked) {
+      inputTask.style.textDecoration = 'line-through';
+    }
+
+    inputTask.addEventListener('change', () => {
+      modifyDescription(index, inputTask.value);
```
Comment (Kamba56) [inline src/index.js]: - [ ] Kindly create a new file in a `modules` folder and move your functions there, then import it to your index.js file as demanded by the second project requirement. This is also best ES6 practice.

**Review by Kamba56 — CHANGES_REQUESTED** [review]
Hi @rebaz36, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Highlights
- [x] All linters passed👏👏👏
- [x] Great use of Github flow👏👏👏

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by rebaz36** [comment]
Thanks for the tip @Kamba56

File: src/index.js [hunk 4]
```diff
@@ -1,20 +1,88 @@
...
+    completed.addEventListener('click', () => {
+      modificarBox(index, completed.checked, myToDolist);
+
+      if (completed.checked) {
+        inputTask.style.textDecoration = 'line-through';
+      } else {
+        inputTask.style.textDecoration = 'none';
+      }
+    });
+
```
Comment (mahbubul14) [inline src/index.js]: - [ ] As the previous reviewer suggested:

> Kindly create a new file in a modules folder and move your functions there, then import it to your index.js file as demanded by the second project requirement. This is also the best ES6 practice.

- [ ] Please work on it. :+1:

**Review by mahbubul14 — CHANGES_REQUESTED** [review]
Hi @rebaz36 ,

Great work on making the changes requested by the previous reviewer 👏

You've done well implementing some of the requested changes, but there are still some which aren't been addressed yet.

# Required Changes ♻️

- [ ] Kindly check the inline comments under the review. :arrow_down: 

###  To Highlights
:heavy_check_mark: No linter error.
:heavy_check_mark: Proper Github flow used.

### Optional suggestions

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better.

- [ ]  Currently, your `README` file is not descriptive and professional enough. Please update your `README` file. You can use [this template ](https://github.com/microverseinc/readme-template)for the `README` but remember to customize it to your project.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification. You can also reach me on [slack](https://microverse-students.slack.com/team/U01J39GL614)

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/index.js [hunk 5]
```diff
@@ -1,20 +1,81 @@
...
```
Comment (franclobo) [inline src/index.js]: - [ ] As the last code reviewer said 👀: **Kindly create a new file in a modules folder and move your functions there, then import it to your index.js file as demanded by the second project requirement. This is also the best ES6 practice.**

**Review by franclobo — CHANGES_REQUESTED** [review]
Hi @rebaz36, 

# Required Changes ♻️

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.


 ## Highlights

- No linter errors ✔️ 
- Correct Github flow ✔️ 
- Professional README file ✔️ 
- Descriptive PR ✔️ 

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
**_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._**

**Review by fobadara — APPROVED** [review]
Hi @rebaz36, 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## Highlights
✅ Linters are passing

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**